api/v1alpha1/serviceintentions_types.go

```diff
@@ -296,12 +333,34 @@ func (in IntentionPermissions) validate(path *field.Path) field.ErrorList {
 
 func (in *IntentionHTTPPermission) validate(path *field.Path) field.ErrorList {
 	var errs field.ErrorList
+	if (in.PathExact != "" && in.PathPrefix != "") || (in.PathExact != "" && in.PathRegex != "") || (in.PathPrefix != "" && in.PathRegex != "") {
+		errs = append(errs, field.Invalid(path, in, `At most only one of pathExact, pathPrefix, or pathRegex may be configured.`))
+	}
 	if invalidPathPrefix(in.PathPrefix) {
 		errs = append(errs, field.Invalid(path.Child("pathPrefix"), in.PathPrefix, `must begin with a '/'`))
 	}
 	if invalidPathPrefix(in.PathExact) {
 		errs = append(errs, field.Invalid(path.Child("pathExact"), in.PathExact, `must begin with a '/'`))
 	}
+	for i, method := range in.Methods {
+		methods := []string{"GET", "HEAD", "POST", "PUT", "DELETE", "CONNECT", "OPTIONS", "TRACE", "PATCH"}
+		if !sliceContains(methods, strings.ToUpper(method)) {
```

Comment:
does consul also not care about casing?

Reply:
Checked consul. It does want everything to be uppercase. Fixing the validation.